### PR TITLE
A2b project la data enrichment

### PR DESCRIPTION
**Pull request by jordan-beard**
Implemented a background service used for enriching the project record.
It is set to run every minute by default (though this is configurable).
This change will introduce a dependency on the academies API. Though the solution should be robust enough to continue working if that API is available.

### CODE REVIEW

**Review by paullocknimble — APPROVED** [review]

File: Dfe.Academies.Academisation.Data.UnitTest/ProjectAggregate/IncompleteProjectsGetDataQueryTests.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+			_subject = new IncompleteProjectsGetDataQuery(_context);
+		}
+
+		[Fact]
+		public async Task ProjectExists___GetProject()
+		{
+			// arrange
+			(ProjectDetails projectDetails1, ProjectState projectState1) = CreateTestProject(region: "region1");
+			(ProjectDetails projectDetails2, ProjectState projectState2) = CreateTestProject("localAuth2");
+			(_, ProjectState projectState3) = CreateTestProject("localAuth3", "region3");
```
Comment (dneed-nimble) [inline Dfe.Academies.Academisation.Data.UnitTest/ProjectAggregate/IncompleteProjectsGetDataQueryTests.cs:32]: More for me to understand, why is the discard used in this test? I assume to denote that the project does not exist and should not be returned? Might be worth having a comment or something for the slower people like me :D

File: Dfe.Academies.Academisation.IData/Establishment/Establishment.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+﻿namespace Dfe.Academies.Academisation.IData.Establishment
+{
+	public class Establishment
+	{
+		public Establishment()
+		{
+			Gor = new Region();
```
Comment (dneed-nimble) [inline Dfe.Academies.Academisation.IData/Establishment/Establishment.cs:7]: Do we want to use the name Gor? If so what does it mean?

File: Dfe.Academies.Academisation.Service.UnitTest/Commands/Legacy/Project/EnrichProjectCommandTests.cs [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+using Dfe.Academies.Academisation.Service.Commands.Legacy.Project;
+using Microsoft.Extensions.Logging;
+using Moq;
+using Xunit;
+
+namespace Dfe.Academies.Academisation.Service.UnitTest.Commands.Legacy.Project
+{
+	public class EnrichProjectCommandTests
+	{
+		private readonly EnrichProjectCommand _sut;
```
Comment (dneed-nimble) [inline Dfe.Academies.Academisation.Service.UnitTest/Commands/Legacy/Project/EnrichProjectCommandTests.cs:19]: What does _sut stand for?

File: Dfe.Academies.Academisation.Service.UnitTest/Commands/Legacy/Project/EnrichProjectCommandTests.cs [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+			var project = _fixture.Create<Domain.ProjectAggregate.Project>();
+			_incompleteProjectGetDataQuery.Setup(m => m.GetIncompleteProjects())
+				.ReturnsAsync(new List<IProject> { project });
+
+			_establishmentGetDataQuery.Setup(m => m.GetEstablishment(It.IsAny<int>()))
+				.ReturnsAsync(_fixture.Create<Establishment>());
+
+			await _sut.Execute();
+
+			Assert.Multiple(
```
Comment (dneed-nimble) [inline Dfe.Academies.Academisation.Service.UnitTest/Commands/Legacy/Project/EnrichProjectCommandTests.cs:86]: Do you need to Asser Multiple here?

File: Dfe.Academies.Academisation.Data.UnitTest/ProjectAggregate/IncompleteProjectsGetDataQueryTests.cs (hunk 1 above)
Comment (jordan-beard) [reply Dfe.Academies.Academisation.Data.UnitTest/ProjectAggregate/IncompleteProjectsGetDataQueryTests.cs:32]: Project 3 has a region and local authority which means it complete and not required to be changed.
Perhaps I could rename it to complete project? Very open to suggestions on how to make this clearer.

File: Dfe.Academies.Academisation.Service.UnitTest/Commands/Legacy/Project/EnrichProjectCommandTests.cs (hunk 3 above)
Comment (jordan-beard) [reply Dfe.Academies.Academisation.Service.UnitTest/Commands/Legacy/Project/EnrichProjectCommandTests.cs:19]: system under test
Can change to subject if that's clearer?

File: Dfe.Academies.Academisation.IData/Establishment/Establishment.cs (hunk 2 above)
Comment (dneed-nimble) [reply Dfe.Academies.Academisation.IData/Establishment/Establishment.cs:7]: If it's to match a property, I think its probably fine, more curiosity as to where it comes from.

File: Dfe.Academies.Academisation.Service.UnitTest/Commands/Legacy/Project/EnrichProjectCommandTests.cs (hunk 4 above)
Comment (jordan-beard) [reply Dfe.Academies.Academisation.Service.UnitTest/Commands/Legacy/Project/EnrichProjectCommandTests.cs:86]: Good point. No. Will change it.

File: Dfe.Academies.Academisation.Service.UnitTest/Commands/Legacy/Project/EnrichProjectCommandTests.cs (hunk 3 above)
Comment (dneed-nimble) [reply Dfe.Academies.Academisation.Service.UnitTest/Commands/Legacy/Project/EnrichProjectCommandTests.cs:19]: No need, now I know the above it makes sense.

**Review by dneed-nimble — APPROVED** [review]

File: Dfe.Academies.Academisation.Service.UnitTest/Commands/Legacy/Project/EnrichProjectCommandTests.cs (hunk 4 above)
Comment (jordan-beard) [reply Dfe.Academies.Academisation.Service.UnitTest/Commands/Legacy/Project/EnrichProjectCommandTests.cs]: This is done.

File: Dfe.Academies.Academisation.Data.UnitTest/ProjectAggregate/IncompleteProjectsGetDataQueryTests.cs (hunk 1 above)
Comment (jordan-beard) [reply Dfe.Academies.Academisation.Data.UnitTest/ProjectAggregate/IncompleteProjectsGetDataQueryTests.cs]: Have updated the variable names to make this clearer.

**Review by dneed-nimble — APPROVED** [review]

**Pull request merged**